src/stacks-hierarchy/saved-payment-utils.ts

```diff
@@ -3,17 +3,32 @@ import {SavedPaymentOption} from './types';
 import {storage} from '@atb/storage';
 import Bugsnag from '@bugsnag/react-native';
 import {useAuthState} from '@atb/auth';
+import {useListRecurringPaymentsQuery} from '@atb/ticketing/use-list-recurring-payments-query';
 
 export function usePreviousPaymentMethod(): SavedPaymentOption | undefined {
   const [paymentMethod, setPaymentMethod] = useState<
     SavedPaymentOption | undefined
   >(undefined);
   const {userId} = useAuthState();
 
+  const {data: recurringPayments} = useListRecurringPaymentsQuery();
+
   useEffect(() => {
     async function run(uid: string) {
-      const method = await getPreviousPaymentMethodByUser(uid);
-      setPaymentMethod(method);
+      const savedMethod = await getPreviousPaymentMethodByUser(uid);
+      if (
+        recurringPayments &&
+        savedMethod &&
+        savedMethod.savedType === 'recurring'
+      ) {
+        const method = recurringPayments.find(
+          (recurringPayment) =>
+            recurringPayment.id === savedMethod.recurringCard.id,
+        );
+        if (method) setPaymentMethod(savedMethod);
+      } else {
+        setPaymentMethod(savedMethod);
```

Comment:
@gorandalum I pushed the change, should now have the same functionality as before.